### PR TITLE
8.3.0: New setting: Regex vs. 'contains' search for Autorecs

### PR DESCRIPTION
**Pull request by ksooo**
Feature suggested in the Forum: https://forum.kodi.tv/showthread.php?tid=361542

screenshot00002

### CODE REVIEW

File: src/tvheadend/AutoRecordings.cpp [hunk 1]
```diff
@@ -174,8 +175,15 @@ PVR_ERROR AutoRecordings::SendAutorecAddOrUpdate(const kodi::addon::PVRTimer& ti
...
   htsmsg_add_str(m, "name", timer.GetTitle().c_str());
 
-  /* epg search data match string (regexp) */
-  htsmsg_add_str(m, "title", timer.GetEPGSearchString().c_str());
+  /* epg search data match string */
+  std::string searchString = timer.GetEPGSearchString();
+  if (!Settings::GetInstance().GetAutorecUseRegEx())
+  {
+    // escape regex special chars
+    static const std::regex specialChars(R"([-[\]{}()*+?.,\^$|#])");
```
Comment (Jalle19) [inline src/tvheadend/AutoRecordings.cpp:183]: Have you validated that this regular expression does what you want it to do?

Comment (ksooo) [reply src/tvheadend/AutoRecordings.cpp:183]: Yes. Do you have concerns?

Comment (ksooo) [reply src/tvheadend/AutoRecordings.cpp:183]: I found it on stackoverflow.

Comment (Jalle19) [reply src/tvheadend/AutoRecordings.cpp:183]: I had some, but when I looked closer I think they're irrelevant. Mainly about escaping `-`, but I think it's okay in its current form.

**Review by Jalle19 — APPROVED** [review]

File: src/tvheadend/AutoRecordings.cpp (hunk 1 above)
Comment (ksooo) [reply src/tvheadend/AutoRecordings.cpp:183]: You may be right. Shall I remove `-` from the list?

Comment (ksooo) [reply src/tvheadend/AutoRecordings.cpp:183]: And what about the `,`?

Comment (ksooo) [reply src/tvheadend/AutoRecordings.cpp:183]: Reading a bit further, I think I will let it like it is now.

**Pull request merged**